### PR TITLE
Adding odf rules

### PR DESCRIPTION
**Pull request by liranmauda**
Adding odf rules

Signed-off-by: liranmauda <liran.mauda@gmail.com>

### CODE REVIEW

**Comment by liranmauda** [comment]
@anmolsachan Can you have a look?

File: rules/rules.libsonnet [hunk 1]
```diff
@@ -56,6 +56,51 @@
...
+            ||| % $._config,
+            labels: {           
+                system_vendor: 'Red Hat',
+                system_type: 'OCS',
+            },
+          },
+          {
+            record: 'job:odf_system_raw_capacity_used_bytes',
+            expr: |||
+              NooBaa_system_capacity
```
Comment (nimrod-becker) [inline rules/rules.libsonnet:75]: Do we want this or the sum of the providers logical?

File: rules/rules.libsonnet [hunk 2]
```diff
@@ -56,6 +56,51 @@
           },
         ],
       },
+      {
+        name: 'noobaa-odf.rules',
+        rules: [
+          {
+            record: 'odf_system_health_status',
```
Comment (nimrod-becker) [inline rules/rules.libsonnet:63]: Don't we need to change the values of 1 and 2 and their mappings? Or are they already aligned?

File: rules/rules.libsonnet [hunk 3]
```diff
@@ -56,6 +56,51 @@
...
+            expr: |||
+              NooBaa_health_status
+            ||| % $._config,
+            labels: {           
+                system_vendor: 'Red Hat',
+                system_type: 'OCS',
+            },
+          },
+          {
+            record: 'job:odf_system_raw_capacity_used_bytes',
```
Comment (anmolsachan) [inline rules/rules.libsonnet:73]:
```suggestion
            record: 'odf_system_raw_capacity_used_bytes',
```

File: rules/rules.libsonnet [hunk 4]
```diff
@@ -56,6 +56,51 @@
...
+            expr: |||
+              NooBaa_system_capacity
+            ||| % $._config,
+            labels: {           
+                system_vendor: 'Red Hat',
+                system_type: 'OCS',
+            },
+          },
+          {
+            record: 'job:odf_system_iops_total_bytes:sum',
```
Comment (anmolsachan) [inline rules/rules.libsonnet:83]:
```suggestion
            record: 'odf_system_iops_total_bytes',
```

File: rules/rules.libsonnet [hunk 5]
```diff
@@ -56,6 +56,51 @@
...
+            expr: |||
+              sum(NooBaa_providers_ops_read_num + NooBaa_providers_ops_write_num )
```
Comment (anmolsachan) [inline rules/rules.libsonnet:85]:
```suggestion
              sum by (namespace, managedBy, job, service) (rate(NooBaa_providers_ops_read_num[1m]) + rate(NooBaa_providers_ops_write_num[1m]))
```

File: rules/rules.libsonnet [hunk 6]
```diff
@@ -56,6 +56,51 @@
...
+            ||| % $._config,
+            labels: {
+                system_vendor: 'Red Hat',
+                system_type: 'OCS',
+            },
+          },
+          {
+            record: 'job:odf_system_throughput_total_bytes:sum',
```
Comment (anmolsachan) [inline rules/rules.libsonnet:93]:
```suggestion
            record: 'odf_system_throughput_total_bytes',
```

File: rules/rules.libsonnet [hunk 7]
```diff
@@ -56,6 +56,51 @@
...
+            expr: |||
+              sum(NooBaa_providers_bandwidth_write_size + NooBaa_providers_bandwidth_read_size)
```
Comment (anmolsachan) [inline rules/rules.libsonnet:95]:
```suggestion
              sum by (namespace, managedBy, job, service) (rate(NooBaa_providers_bandwidth_read_size[1m]) + rate(NooBaa_providers_bandwidth_write_size[1m]))
```

**Review by anmolsachan — CHANGES_REQUESTED** [review]

**Comment by anmolsachan** [comment]
IOPS and Throughput according to my queries : 

IOPS
Screenshot from 2021-08-25 17-10-33

Throughput
Screenshot from 2021-08-25 17-11-00

File: rules/rules.libsonnet (hunk 2 above)
Comment (liranmauda) [reply rules/rules.libsonnet:63]: https://github.com/noobaa/noobaa-core/blob/48d27e51a5e61ea1a5d7f77aa6c3063f12723e67/src/server/system_services/stats_aggregator.js#L977-L980

so we need to aligned in core...

Comment (nimrod-becker) [reply rules/rules.libsonnet:63]: that would change the ui ... cant we change in rule only?

Comment (liranmauda) [reply rules/rules.libsonnet:63]: We will expose new metrics that return what we need in core. I will put the PR here for refrance.

File: rules/rules.libsonnet (hunk 1 above)
Comment (liranmauda) [reply rules/rules.libsonnet:75]: I am not sure, I think that this is the most fitting one.

File: rules/rules.libsonnet (hunk 2 above)
Comment (liranmauda) [reply rules/rules.libsonnet:63]: https://github.com/noobaa/noobaa-core/pull/6718

Comment (liranmauda) [reply rules/rules.libsonnet:63]: Now that is in We are calling `NooBaa_odf_health_status`

**Review by nimrod-becker — APPROVED** [review]

**Pull request merged**